py_ecc/fields/field_elements.py

```diff
@@ -187,26 +181,23 @@ class FQP(object):
     A class for elements in polynomial extension fields
     """
     degree = 0
-    curve_name = None
     field_modulus = None
 
     def __init__(self,
                  coeffs: Sequence[IntOrFQ],
                  modulus_coeffs: Sequence[IntOrFQ]=None) -> None:
-        if self.curve_name is None:
-            raise AttributeError("Curve Name hasn't been specified")
         if self.field_modulus is None:
             raise AttributeError("Field Modulus hasn't been specified")
 
         if len(coeffs) != len(modulus_coeffs):
             raise Exception(
                 "coeffs and modulus_coeffs aren't of the same length"
             )
-        # Encoding all coefficients in type FQ (in regards to the curve name too)
+        # Encoding all coefficients in the corresponding type FQ
         self.FQP_corresponding_FQ_class = type(
```

Comment:
Not for this PR but, This appears to be a class-level value (i.e. it is constant across all instances of this class since it only differs on the `field_modulus` property.).

If that is the case, then it *should* be possible to populate this at the class level either by doing it manually *or* by doing it using a metaclass.  Rough untested code looks something like:


```python
class FQPMetaclass(type):
    def __new__(cls, name, bases, attrs):
        field_modulus = attrs.get('field_modulus', cls.field_modulus)
        if field_modulus is not None:
            attrs['FQP_corresponding_FQ_class'] = type("FQP_corresponding_FQ_class", (FQ,), {'field_modulus": field_modulus})
        return super().__new__(name, bases, attrs)
```

Reply:
Opened #62 for this.